internal/imapwire/decoder.go

```diff
@@ -133,9 +133,13 @@ func (dec *Decoder) Expect(ok bool, name string) bool {
 	return true
 }
 
+// SP consumes all following spaces and returns true if there followed by more data.
+// Note, it may still consume spaces (trailing ones) even if it returns false.
 func (dec *Decoder) SP() bool {
 	if dec.acceptByte(' ') {
-		return true
+		for dec.acceptByte(' ') {
+		}
+		return !dec.endOfLine()
```

Comment:
Can we add a `// https://github.com/emersion/go-imap/issues/571` comment to explain why this is done here?